.github/workflows/scripts/scanner-fetch-nvd.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+base_url="https://services.nvd.nist.gov/rest/json/cves/2.0"
+# NVD CVE Api has limit of 2000 CVEs per page at maximum
+results_per_page=2000
+dir_name="nvd-data"
+
+# Function to fetch data for a given period
+fetch_data_for_period () {
+  local period_start_date=$1
+  local period_end_date=$2
+  local current_start_index=0
+  local total_results=0
+  local file_name
+  local success
+
+  while (( current_start_index < total_results || total_results == 0 )); do
+    file_name="${period_start_date}_${current_start_index}.json"
+    echo "Fetching data from ${period_start_date} to ${period_end_date}, starting at index ${current_start_index}..."
+
+    retry_intervals=(2 4 8) # Array of retry intervals in seconds
+    success=false
+
+    for i in "${retry_intervals[@]}"; do
+        if curl -s --fail --show-error -H "apiKey: $API_KEY" "$base_url?pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
```

Comment:
```suggestion
        if curl -s --fail --show-error -H "apiKey: $API_KEY" "$base_url?noRejected&pubStartDate=$period_start_date&pubEndDate=$period_end_date&startIndex=$current_start_index" -o "$dir_name/$file_name"; then
```

that way we won't download rejected vulns, and the total number of vulns we get is reduced

Reply:
I'm also not a `curl` expert, so I'm wonder/hoping this doesn't expose the API key, does it? How about upon error?

Reply:
Sorry some content of the script was created before the ADR was merged. I should have added that.